client-app/shared/catalog/components/product/variations-table.vue

```diff
@@ -0,0 +1,279 @@
+                  </template>
+                </VcTooltip>
+              </template>
+            </VcQuantity>
+
+            <CountInCart :product-id="variation.id" class="variations-table__in-cart" />
+          </td>
+        </tr>
+      </template>
+    </VcTable>
+  </div>
+</template>
+
+<script setup lang="ts">
+import _ from "lodash";
+import { ref, computed, onMounted } from "vue";
+import { useI18n } from "vue-i18n";
+import { PropertyType } from "@/core/api/graphql/types";
+import { useErrorsTranslator } from "@/core/composables";
+import { MAX_DISPLAY_IN_STOCK_QUANTITY } from "@/core/constants";
+import { getPropertyValue, getPropertiesGroupedByName } from "@/core/utilities";
+import { useShortCart } from "@/shared/cart/composables";
+import CountInCart from "../count-in-cart.vue";
+import type { Product, ShortLineItemFragment, VariationType } from "@/core/api/graphql/types";
+import type { ErrorType } from "@/core/composables";
+import type { NamedValue } from "vue-i18n";
+
+interface IProps {
+  product: Product;
+}
+
+interface IProperties {
+  name: string;
+  label: string;
+  values: string[];
+}
+
+const props = defineProps<IProps>();
+
+const { t } = useI18n();
+
+const properties = ref<IProperties[]>([]);
+const propTitles = ref<ITableColumn[]>([]);
+
+const { cart, addToCart, changeItemQuantity } = useShortCart();
+
+const variations = computed(() => [props.product, ...props.product.variations]);
+
+const columns = computed<ITableColumn[]>(() => [
+  {
+    id: "item",
+    title: t("shared.catalog.product_details.variations.columns.item"),
+    sortable: true,
+    classes: "min-w-52",
+  },
+  ...propTitles.value,
+  {
+    id: "stock",
+    title: t("shared.catalog.product_details.variations.columns.stock"),
+    align: "center",
+    classes: "min-w-[4.375rem] w-[4.375rem]",
+  },
+  {
+    id: "price",
+    title: t("shared.catalog.product_details.variations.columns.price"),
+    align: "right",
+    classes: "min-w-32 w-32",
+  },
+  {
+    id: "quantity",
+    title: t("shared.catalog.product_details.variations.columns.quantity"),
+    sortable: true,
+    align: "center",
+    classes: "min-w-32 w-32",
+  },
+]);
+
+const validationErrors = computed<ErrorType[]>(() => {
+  return (
+    cart.value?.validationErrors.map((el) => {
+      return {
+        id: el.objectId,
+        code: el.errorCode,
+        parameters: el.errorParameters?.reduce((acc, err) => {
+          acc[err.key] = err.value;
+          return acc;
+        }, {} as NamedValue),
+        description: el.errorMessage,
+      };
+    }) || []
+  );
+});
+
+const { idErrors } = useErrorsTranslator("validation_error", validationErrors);
+
+function getTableProperties() {
+  if (!variations.value.length) {
+    return;
+  }
+
+  const propertiesCombined = _.flatten(variations.value.map((variation) => getProperties(variation)));
+
+  const names = _.uniqBy(
+    propertiesCombined.map((prop) => {
+      return {
+        name: prop!.name,
```

Comment:
prop!.name -> props.name
prop!.label! -> prop.label

label could be props defaults